### PR TITLE
task/wg-75-project-listing-main

### PR DESCRIPTION
**Pull request by sophia-massie**
## Overview: ##
Create React project listing.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [WG-75](https://tacc-main.atlassian.net/browse/WG-75)

## Summary of Changes: ##
- Installed core-styles lts and added to index.tsx
- imported styles into index.html head
- Created project listing component and added component to main menu page to view
- created hooks for getting projects 

- created types for projects responses, ds project responses, and ds project collections (raw response)

## Testing Steps: ##
1. Go to localhost:4200 and make sure your projects display
2. Links and button do not work 

## UI Photos:
Screenshot 2024-01-12 at 4 54 14 PM

## Notes: ##
TODOs
- need to create a hook that uses a query function to combine ds response and tapis project response
- need to install icons into project to show the edit and trash icons for each map project

### CODE REVIEW

**Review by tjgrafft — COMMENTED** [review]
Tested locally and I see my correct project listings! Just one comment. When I load the page the ProjectListing component renders the first column 'Map' and its values right away, but the 'Project' column's values render after a slight delay. If I refresh the page the delay is still there. Looking at the network tab it takes 0.83 ms for the '/projects' request, while the /v2 request is taking 1.4 seconds. This could be causing the delay. Not sure, but maybe you can synchronize the rendering of the Map and Project columns by controlling their visibility based on the loading state of the asynchronous data

**Comment by sophia-massie** [comment]
> Tested locally and I see my correct project listings! Just one comment. When I load the page the ProjectListing component renders the first column 'Map' and its values right away, but the 'Project' column's values render after a slight delay. If I refresh the page the delay is still there. Looking at the network tab it takes 0.83 ms for the '/projects' request, while the /v2 request is taking 1.4 seconds. This could be causing the delay. Not sure, but maybe you can synchronize the rendering of the Map and Project columns by controlling their visibility based on the loading state of the asynchronous data

Yes, definitely good catch. That is part of the follow on work I have to combine the response and track the state of both responses so that the table doesn't load until all the data is there.

**Review by tjgrafft — APPROVED** [review]

File: react/src/components/Projects/ProjectListing.tsx
Comment (nathanfranklin) [inline react/src/components/Projects/ProjectListing.tsx:26]: "maps" is preferred over "projects" in what the user sees on frontend (even though we use that term in Geoapi). Can we match what's currently in the angular client ([here](https://github.com/TACC-Cloud/hazmapper/blob/master/angular/src/app/components/main-welcome/main-welcome.component.html#L39C24-L44)) like the "Failed to Connect to Server" and possible error information? We added more details recently (see https://github.com/TACC-Cloud/hazmapper/pull/148) regarding the error as some accounts hit some strange behavior and helps us doing support.

File: react/src/components/Projects/ProjectListing.tsx
Comment (nathanfranklin) [inline react/src/components/Projects/ProjectListing.tsx:1]: Can a unit test for this be added?

**Review by nathanfranklin — CHANGES_REQUESTED** [review]
Looks good 👍  

 Here are some additional comments.  I also left some follow on ui-polishing/functionality on [slack](https://tacc-team.slack.com/archives/C04T8CKSWP5/p1705617510953699?thread_ts=1705100215.274389&cid=C04T8CKSWP5)

File: react/src/types/projects.ts
Comment (nathanfranklin) [reply react/src/types/projects.ts:22]: new `transform` param would be a way to change the result and drop the `value` https://github.com/TACC-Cloud/hazmapper/blob/master/react/src/hooks/user/useAuthenticatedUser.ts#L10-L12

File: react/src/hooks/projects/useProjects.ts
Comment (taoteg) [reply react/src/hooks/projects/useProjects.ts:18]: second that =)

**Review by taoteg — COMMENTED** [review]
So far, looks good. 
Waiting for the edits Nathan requested before final review and approval.

**Review by nathanfranklin — APPROVED** [review]
LGTM 👍

**Pull request merged**
